lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -640,8 +640,10 @@ bool SPIRVRegularizeLLVMBase::regularize() {
         if (auto *ASCast = dyn_cast<AddrSpaceCastInst>(&II)) {
           Type *DestTy = ASCast->getDestTy();
           Type *SrcTy = ASCast->getSrcTy();
-          if (DestTy->getPointerElementType() !=
-              SrcTy->getPointerElementType()) {
+          if (!II.getContext().supportsTypedPointers())
+            continue;
+          if (DestTy->getNonOpaquePointerElementType() !=
+              SrcTy->getNonOpaquePointerElementType()) {
```

Comment:
`if (!DestTy->hasSameElementTypeAs(SrcTy))` should avoid explicitly checking for typed pointers here.

Reply:
@jcranmer-intel this comment hasn't been addressed yet, other than that this PR looks good to me.